src/lua_sysctl.c

```diff
@@ -514,10 +514,10 @@ luaA_sysctl_get(lua_State *L)
 				else
 					lua_pushinteger(L, mv);
 			} else {
-				if (intlen > sizeof(lua_Unsigned))
+				if (intlen > sizeof(lua_Integer))
```

Comment:
Now that I think of it, this condition is incorrect.

An `unsigned int` may hold a value that cannot be represented as a `signed int` (e.g. `INT_MAX + 1`). Thus, checking the representation length doesn't guarantee the cast from `unsigned int` to `signed int` is safe. Instead, the value should be checked:
```suggestion
				if (umv > LUA_MAXINTEGER)
```

Unfortunately, lua 5.2 doesn't define `LUA_MAXINTEGER` so we have to add some compat goo after the lua includes:
```c
/*
 * lua 5.3+ define LUA_MAXINTEGER, but lua 5.2 does not. See
 * https://www.lua.org/manual/5.2/manual.html#lua_Integer
 */
#ifndef LUA_MAXINTEGER
#define LUA_MAXINTEGER  PTRDIFF_MAX
#endif
```